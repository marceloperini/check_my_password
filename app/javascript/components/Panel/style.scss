.Panel {
  width: 100%;
  background-color: white;
  border-radius: 8px;
  padding: 16px;
  box-shadow: 3px 3px 8px rgba(0, 0, 0, 0.04);
}
