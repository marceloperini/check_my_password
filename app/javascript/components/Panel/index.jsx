import PropTypes from 'prop-types';
import React from 'react';

import './style.scss';

const Panel = (props) => {
  const { children } = props;

  return (
    <div className='Panel'>
      { children }
    </div>
  );
};

Panel.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Panel;
