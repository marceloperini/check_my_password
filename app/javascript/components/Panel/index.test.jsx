import React from 'react';
import renderer from 'react-test-renderer';

import Panel from '.';

describe('Panel', () => {
  test('with children prop', () => {
    const component = renderer.create(
      <Panel>
        <h1>Testing My Code!</h1>
      </Panel>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
