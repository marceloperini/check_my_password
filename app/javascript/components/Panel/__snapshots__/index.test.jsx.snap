// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel with children prop 1`] = `
<div
  className="Panel"
>
  <h1>
    Testing My Code!
  </h1>
</div>
`;
